Replace openrc commands by their systemd equivalents

Patch by Gordon Bos

--- a/admin/helpers/exec_helper.php
+++ b/admin/helpers/exec_helper.php
@@ -32,41 +32,22 @@
 
 function invoke_rc_d( $name, $action ) {
     $cmd = array(
-        "/sbin/rc-service", 
-        "-q", 
-        $name, 
-        $action
+        "/usr/bin/systemctl",
+        "-q",
+        $action,
+        $name
     );
     exec( escapeshellargs( $cmd ), $output, $retval );
     return $retval == 0;
 }
 
-function update_rc_d( $name, $action="defaults", $priority=0, $runlevel=0) {
-	if($action == "enable") {
-		$cmd = array(
-			"/sbin/rc-update",
-			"-q",
-			"add",
-			$name,
-			"default"
-		);
-	} elseif($action == "disable") {
-		$cmd = array(
-			"/sbin/rc-update",
-			"-q",
-			"del",
-			$name,
-			"default"
-		);
-	} else {
-		$cmd = array(
-			"/sbin/rc-update",
-			"-q",
-			$action,
-			$name,
-			"default"
-		);
-	}
+function update_rc_d( $name, $action="enable", $priority=0, $runlevel=0) {
+	$cmd = array(
+		"/usr/bin/systemctl",
+		"-q",
+		$action,
+		$name
+	);
 	exec( escapeshellargs( $cmd ), $output, $retval );
 	return $retval == 0;
 }
--- a/admin/models/networkmanager.php
+++ b/admin/models/networkmanager.php
@@ -164,8 +164,7 @@
     if (!$newdomain) {
         $newdomain = $olddomain? $olddomain:"localdomain";
     }
-    file_put_contents('/etc/conf.d/hostname', "hostname=\"$newname\"");
-    _system('rc-config', 'restart', 'hostname');
+    _system('/usr/bin/hostnamectl', 'set-hostname', $newname);
 
     $lanip = $this->_get_ip($this->get_lan_interface());
     $oldhosts = _system("grep -m 1 \"^".str_replace('.','\.',$lanip)."\s\" /etc/hosts");
